python/ray/data/datasource/image_datasource.py

```diff
@@ -169,14 +90,11 @@ def _read_file(
         if mode is not None:
             image = image.convert(mode)
 
-        label = _get_class_from_path(path, root)
+        builder = ArrowBlockBuilder()
```

Comment:
Could we use `DelegatingBlockBuilder` here, so the Arrow representation is hidden as an implementation detail in the blocker layer? The contract should be that we give `DelegatingBlockBuilder` a NumPy ndarray row, and it decides how to most efficiently represent it as a `Block`.

https://github.com/ray-project/ray/blob/master/python/ray/data/_internal/delegating_block_builder.py